### DISCUSSION TITLE
PAGImageView是否提供cancel/stop功能

### DISCUSSION THREAD

**Discussion by carleft** [Tencent/libpag#1878] [General] [answered]
查看API只提供了pause，且功能描述说再次调用play()会按照上次暂定的进度继续播放
```java
/**
  * Cancels the animation at the current position. Calling the play() method can resume the
  * animation from the last paused position.
  */
public void pause() {
    animator.cancel(); 
}
```

动画监听中提到了关于stop方法的存在，但是搜索不到该方法
```java
        /**
         * Notifies the cancellation of the animation. It can be called from either the UI thread or
         * the thread that calls the **stop()** method.
         */
        void onAnimationCancel(PAGImageView view);
```
请问当View不再被使用，是否需要手动停止动画并释放资源，例如View onDetachedFromWindow回调

**Comment by carleft** [comment]
版本4.3.33 Android

**Comment by kevingpqi123** [comment, marked as answer]
基础的问题建议自己多跑跑 pag 的 demo 工程，当你这边不再需要动画的是直接 pause 就可以，随着 View 的释放，相关资源会自动释放
<img width="737" alt="企业微信截图_df87ba96-d057-4b23-8b60-95f0c617cdef" src="https://github.com/Tencent/libpag/assets/16609221/cb5d2b6a-9280-48db-a4b0-2ba34172dd01">
